titus-server-master/src/main/java/com/netflix/titus/master/jobmanager/service/JobManagerUtil.java

```diff
@@ -148,6 +149,15 @@ public static Task attachPlacementData(Task task, Optional<TitusExecutorDetails>
         }).orElse(task);
     }
 
+    public static Task attachKubeletData(Task task, PodWrapper podWrapper) {
+        String ipv4 = podWrapper.getV1Pod().getStatus().getPodIP();
```

Comment:
You should check defensively that status is not null as well.

Reply:
Good idea

Reply:
We should also add `if(podWrapper.getV1Pod().getStatus() == null) { return task; }`